spring-kafka/src/test/java/org/springframework/kafka/listener/KafkaMessageListenerContainerTests.java

```diff
@@ -540,10 +551,20 @@ public void testBatchAck() throws Exception {
 			for (Entry<TopicPartition, OffsetAndMetadata> entry : map.entrySet()) {
 				if (entry.getValue().offset() == 2) {
 					firstBatchLatch.countDown();
-					latch.countDown();
 				}
 			}
-			return invocation.callRealMethod();
+			Object result = null;
+			try {
+				result = invocation.callRealMethod();
+			}
+			finally {
+				for (Entry<TopicPartition, OffsetAndMetadata> entry : map.entrySet()) {
+					if (entry.getValue().offset() == 2) {
+						latch.countDown();
+					}
+				}
+			}
+			return result;
```

Comment:
Well, actually this can be simplified with direct `return` in the `try`. The `finally` is invoked before exit from the code block: http://www.programmerinterview.com/index.php/java-questions/will-finally-run-after-return/